source/DataFile.cpp

```diff
@@ -0,0 +1,55 @@
+#include "DataFile.h"
+#include <cstdlib>
+#include <chrono>
+#include <sstream>
+
+DataFile::DataFile() {
+    // srandom();
+    int pid = random() % 30000 + 300;
```

Comment:
I had to change `random()` to `rand()` to get it compiling on windows. Does `rand()` work on Mac?